### DISCUSSION TITLE
Thanos update process

### DISCUSSION THREAD

**Discussion by archxlith** [thanos-io/thanos#4068] [Questions & Answers]
I was wondering what the update process is for Thanos. I currently have three components (sidecar, store and compact). I'm looking to update but don't see any mention of how this should be done. Do I just bring all the components down and drop in the updated  binary. Any order they should be done in? Was thinking, bring them all down and then start with sidecar and then store then compact.

**Comment by onprem** [comment]
I don't think there is any particular process we promote. Things should work fine in any order you want to upgrade the components as we try to keep the underlying gRPC APIs backward compatible. That being said since we still haven't reached the `v1.0` milestone, there can be some breaking changes but in that case, they will be properly documented in the CHANGELOG.

So overall, your plan sounds okay to me just don't forget to check the CHANGELOG.
